src/simulator/halo_simulator.py

```diff
@@ -54,6 +60,14 @@
 )
 
 
+MAX_ACTIVE_PUBLISHERS = 20
+
+
+class UserInfo(NamedTuple):
```

Comment:
Is this being used?

Reply:
Yes. Please see the defaultdict variable `user_info`.

Reply:
Decided to remove `UserInfo` class and create a `collections.namedtuple` inside the function.